sdk/workloadssapvirtualinstance/Azure.ResourceManager.SapVirtualInstances/src/Generated/Models/ApplicationServerConfiguration.cs

```diff
@@ -0,0 +1,88 @@
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromString("\"foo\"")</term>
+        /// <description>Creates a payload of "foo".</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromObjectAsJson(new { key = "value" })</term>
+        /// <description>Creates a payload of { "key": "value" }.</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromString("{\"key\": \"value\"}")</term>
+        /// <description>Creates a payload of { "key": "value" }.</description>
+        /// </item>
+        /// </list>
+        /// </para>
+        /// </summary>
+        private IDictionary<string, BinaryData> _serializedAdditionalRawData;
+
+        /// <summary> Initializes a new instance of <see cref="ApplicationServerConfiguration"/>. </summary>
+        /// <param name="subnetId"> The subnet id. </param>
+        /// <param name="virtualMachineConfiguration"> Gets or sets the virtual machine configuration. </param>
+        /// <param name="instanceCount"> The number of app server instances. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="subnetId"/> or <paramref name="virtualMachineConfiguration"/> is null. </exception>
+        public ApplicationServerConfiguration(string subnetId, VirtualMachineConfiguration virtualMachineConfiguration, long instanceCount)
+        {
+            Argument.AssertNotNull(subnetId, nameof(subnetId));
+            Argument.AssertNotNull(virtualMachineConfiguration, nameof(virtualMachineConfiguration));
+
+            SubnetId = subnetId;
+            VirtualMachineConfiguration = virtualMachineConfiguration;
+            InstanceCount = instanceCount;
+        }
+
+        /// <summary> Initializes a new instance of <see cref="ApplicationServerConfiguration"/>. </summary>
+        /// <param name="subnetId"> The subnet id. </param>
+        /// <param name="virtualMachineConfiguration"> Gets or sets the virtual machine configuration. </param>
+        /// <param name="instanceCount"> The number of app server instances. </param>
+        /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
+        internal ApplicationServerConfiguration(string subnetId, VirtualMachineConfiguration virtualMachineConfiguration, long instanceCount, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        {
+            SubnetId = subnetId;
+            VirtualMachineConfiguration = virtualMachineConfiguration;
+            InstanceCount = instanceCount;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> Initializes a new instance of <see cref="ApplicationServerConfiguration"/> for deserialization. </summary>
+        internal ApplicationServerConfiguration()
+        {
+        }
+
+        /// <summary> The subnet id. </summary>
+        public string SubnetId { get; set; }
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

Reply:
Thanks @archerzz , I am using tspconfig for generating the csharp code. the above-mentioned example is for autorest which is not getting accepted along with tspconfig.

Can you please share examples for typespec?